morpheus/pipeline/pipeline.py

```diff
@@ -116,21 +132,35 @@ def add_edge(self, start: typing.Union[StreamWrapper, Sender], end: typing.Union
         elif (isinstance(end, Receiver)):
             end_port = end
 
-        # Ensure both are added to this pipeline
-        self.add_node(start_port.parent)
-        self.add_node(end_port.parent)
-
         start_port._output_receivers.append(end_port)
         end_port._input_senders.append(start_port)
 
-        self._graph.add_edge(start_port.parent,
-                             end_port.parent,
-                             start_port_idx=start_port.port_number,
-                             end_port_idx=end_port.port_number)
+        segment_graph = self._segment_graphs[segment_id]
+        segment_graph.add_edge(start_port.parent,
+                               end_port.parent,
+                               start_port_idx=start_port.port_number,
+                               end_port_idx=end_port.port_number)
+
+    def add_segment_edge(self, egress_stage, egress_segment, ingress_stage, ingress_segment, port_pair):
+        egress_edges = self._segments[egress_segment]["egress_ports"]
+        egress_edges.append({
+            "port_pair": port_pair,
+            "input_sender": egress_stage.unique_name,
+            "output_receiver": ingress_stage.unique_name,
+            "receiver_segment": ingress_segment
+        })
+
+        ingress_edges = self._segments[ingress_segment]["ingress_ports"]
+        ingress_edges.append({
+            "port_pair": port_pair,
+            "input_sender": egress_stage.unique_name,
+            "sender_segment": egress_segment,
+            "output_receiver": ingress_stage.unique_name
+        })
 
     def build(self):
         """
-        This function sequentially activates all of the Morpheus pipeline stages passed by the users to execute a
+        This function sequentially activates all the Morpheus pipeline stages passed by the users to execute a
```

Comment:
Is this intentional? To me the sentence is more readable with "of" in it.

Reply:
Its grammatically incorrect the other way.